larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.h

```diff
@@ -0,0 +1,48 @@
+/**
+ *  @file   larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.h
+ *
+ *  @brief  Header file for cheated clustering tool class.
+ *
+ *  $Log: $
+ */
+
+#ifndef LAR_CHEATED_THREE_D_CLUSTERING_TOOL_H
+#define LAR_CHEATED_THREE_D_CLUSTERING_TOOL_H 1
+
+#include "Pandora/Algorithm.h"
+#include "larpandoracontent/LArReclustering/ThreeDReclusteringAlgorithm.h"
+
+namespace lar_content
+{
+
+class ClusteringTool;
+
+class CheatedThreeDClusteringTool : public ClusteringTool
+{
+public:
+
+    CheatedThreeDClusteringTool();
+
+private:
+
+    bool Run(const pandora::Algorithm *const pAlgorithm, std::vector<pandora::CaloHitList*> &newCaloHitListsVector) override;
```

Comment:
I should check again re upstream constraints, but here I effectively kept the syntax similar to other instances of AlgorithmTool Run functions in Pandora. Should we review the  use of this function across Pandora? Or would you like me to look into making a change for this instance specifically?

Reply:
Agreed with @AndyChappell here
If this does not depend on a definition of an upstream function then it could be changed to be safer.  
A review of usage across Pandora is well beyond the scope of this PR.

Reply:
Not very familiar to reference_wrapper, but I have had a look and it seemed to require quite a bit of restructuring upstream, where these lists are filled, compared etc. This seems a bit painful and more at risk of introducing bugs and require lots of testing. Let me know if you think the changes (including to upstream) should be straightforward, and I am possibly overcomplicating it. But, if not, let me know if I should leave pointers or accept making copies. Thanks!

Reply:
Ok, I had a go at restructuring the code. I am not the biggest fan of reference_wrapper now being in a few places upstream too. But, if you don't see particular issues, then I can leave it as is and avoid more manipulation.
This code compiles but needs to be tested, as there are quite a few changes.